aurora/src/test/java/com/market/banica/aurora/mapper/SubscribeMapperTest.java

```diff
@@ -0,0 +1,196 @@
+        subscribeMapper.renderSubscribe(INVALID_REQUEST, responseObserver);
+        verify(responseObserver, times(1)).onError(any());
+    }
+
+    @Test
+    void renderSubscribeWithRequestForOrderBookInvokesOnError() {
+        when(channelManager.getAllChannelsContainingPrefix(any())).thenReturn(Collections.singletonList(dummyManagedChannel));
+        subscribeMapper.renderSubscribe(ORDERBOOK_REQUEST, responseObserver);
+        verify(responseObserver, times(1)).onError(any());
+    }
+
+    @Test
+    void renderSubscribeWithRequestForUnsupportedServiceInvokesOnError() {
+        when(channelManager.getAllChannelsContainingPrefix(any())).thenReturn(Collections.singletonList(dummyManagedChannel));
+        subscribeMapper.renderSubscribe(INVALID_REQUEST, responseObserver);
+        verify(responseObserver, times(1)).onError(any());
+    }
+
+    @Test
+    void renderSubscribeWithRequestForMarketServiceSubscribesResponseObserverAndCallsOnNextAndOnCompleted() throws IOException, InterruptedException {
+        //Arrange
+        when(channelManager.getAllChannelsContainingPrefix(any())).thenReturn(Collections.singletonList(dummyManagedChannel));
+        when(stubManager.getMarketStub(any())).thenReturn(marketStub);
+
+        createFakeMarketReplyingServer();
+        grpcCleanup.register(marketReceiverChannel);
+        marketReceiverChannel.getState(true);
+
+        //Act
+        subscribeMapper.renderSubscribe(MARKET_REQUEST, responseObserver);
+        Thread.sleep(1000);
+
+        //Assert
+        verify(stubManager, times(1)).getMarketStub(any());
+        verify(responseObserver, times(1)).onNext(any());
+        verify(responseObserver, times(1)).onCompleted();
+    }
+
+    @Test
+    void renderSubscribeWithRequestForAuroraServiceSubscribesResponseObserverAndCallsOnNextAndOnCompleted() throws IOException, InterruptedException {
+        //Arrange
+        when(channelManager.getAllChannelsContainingPrefix(any())).thenReturn(Collections.singletonList(dummyManagedChannel));
+        when(stubManager.getAuroraStub(any())).thenReturn(auroraStub);
+
+        createFakeAuroraReplyingServer();
+        grpcCleanup.register(auroraReceiverChannel);
+        auroraReceiverChannel.getState(true);
+
+        //Act
+        subscribeMapper.renderSubscribe(AURORA_REQUEST, responseObserver);
+        Thread.sleep(1000);
+
+        //Assert
+        verify(stubManager, times(1)).getAuroraStub(any());
+        verify(responseObserver, times(3)).onNext(any());
+        verify(responseObserver, times(1)).onCompleted();
+    }
+
+    private void createFakeMarketReplyingServer() throws IOException {
+        grpcCleanup.register(InProcessServerBuilder
+                .forName(marketReceiverName)
+                .directExecutor()
+                .addService(this.fakeReceivingMarketService())
+                .build()
+                .start());
+    }
+
+    private void createFakeAuroraReplyingServer() throws IOException {
```

Comment:
This method is duplication with the one in RequestMapperTest. You can move all createFake...Server methods with the methods that create stubs in a parent class FakeStubsBaseTest or something similar and extend the current classes with it. Other option is to create util class that will be responsible for creating the fake servers and stubs

Reply:
I've created a new util package inside the test/java folder where I've implemented FakeStubsGenerator class to resolve the duplication issue.

Reply:
Can you move there the rest of the fake stubs and servers